src/Diagnostics/Debug/StateCheck.jl

```diff
@@ -0,0 +1,556 @@
+"""
+ StateCheck :: Module with a minimal set of functions for gettings statistics 
+               and basic I/O from ClimateMachine DG state arrays (MPIStateArray type). 
+               Created for regression testing and code change tracking and debugging.
+               StateCheck functions iterate over named variables in an MPIStateArray,
+               calculate and report their statistics and/or write values for all or
+               some subset of points at a fixed frequency.
+
+               Functions:
+               StateCheck.sccreate()   - Create a StateCheck call back variable.
+
+               StateCheck.scdocheck()  - Check Statecheck variable values against reference values.
+
+               StateCheck.scprintref() - Print Statecheck variable in format for creating 
+                                         reference values.
+"""
+module StateCheck
+
+# Imports from standard Julia packages
+using Formatting
+using MPI
+using Printf
+using StaticArrays
+using Statistics
+
+# Imports from ClimateMachine core
+import ClimateMachine.GenericCallbacks: EveryXSimulationSteps
+import ClimateMachine.MPIStateArrays: MPIStateArray
+import ClimateMachine.VariableTemplates: flattenednames
+
+####
+# For testing put a new function signature here!
+# Needs to go in src/Utilities/VariableTemplates/var_names.jl
+# This handles SMatrix case
+flattenednames(::Type{T}; prefix = "") where {T <: SArray} =
+    ntuple(i -> "$prefix[$i]", length(T))
+####
+
+struct vstat
+    max
+    min
+    mean
+    std
+end
+
+# Global functions to expose
+# sccreate - Create a state checker call back
+export sccreate
+export scdocheck
+export scprintref
+
+# ntFreqDef:: default frequency (in time steps) for output.
+# precDef  :: default precision used for formatted output table
+ntFreqDef = 10
+precDef = 15
+
+"""
+ sccreate :: Create a a "state check" call-back for one or more MPIStateArrays  \n
+             that will report basic statistics for the fields in the array.
```

Comment:
The convention is that the first line of the docstring should be the function signature, indented 4 spaces so that it renders as a code block.

Also `::` has special meaning in Julia, so probably not a good idea to use it in other contexts.